### ISSUE TITLE
Unable to mount

### ISSUE THREAD

**Issue by davistan** [veracrypt/VeraCrypt#45]
Hi, after I have successfully created the container, now I can't get it to mount using command line.
```
Error: mount: wrong fs type, bad option, bad superblock on /dev/mapper/veracrypt1,
       missing codepage or helper program, or other error
       In some cases useful info is found in syslog - try
       dmesg | tail  or so
```
and running dmesg returns this:
```
[20954.613645] bio: create slab <bio-1> at 1
[20954.695824] EXT4-fs (dm-1): VFS: Can't find ext4 filesystem
```

any pointers on how to solve this?

**Comment by idrassi** [comment]
You are running under which Linux distribution?
Are you trying to mount a file container or a partition?

**Comment by davistan** [comment]
I'm using ubuntu 14.04. I'm mounting the file container that was created from my previous command line issue #39. Is there extra steps needed before able to mount?

**Comment by initbar** [comment]
I think I might be able to help,

@davistan: reading issue #39 more closely, I've noticed you've used `--filesystem=ext4` to generate the VC container (my apologies for not catching it sooner). However, `man` notes only `FAT` and `none` types are allowed if `--filesystem` flag is indicated (as of VeraCrypt 1.14, Linux):

![scr2](https://cloud.githubusercontent.com/assets/12630151/12539023/2c7243da-c2b7-11e5-902b-a726d274d328.png)

Using this idea, I've conducted following tests to confirm whether the error was triggered by `--filesystem=ext4`:

| Command Variations | Analysis | Error Triggered |
| :--------: | :--------: | :--------: |
| `--filesystem=ext4` | Using `veracrypt -t -c --encryption AES-Twofish --filesystem=ext4 --hash SHA512 --volume-type normal --size 1000000 --random-source /dev/urandom`, VeraCrypt container is created without a hitch. @davistan's error message triggered when mounting using `veracrypt ${container} /media/veracrypt2` | :white_check_mark: |
| `--filesystem=(random)` | Using `veracrypt -t -c --encryption AES-Twofish --filesystem=$(perl -e "print 'A'x65536") --hash SHA512 --volume-type normal --size 1000000 --random-source /dev/urandom`, VeraCrypt container is created without a hitch. Error message triggered when mounting using `veracrypt ${container} /media/veracrypt2` | :white_check_mark: |
| `--filesystem=FAT` | Using `veracrypt -t -c --encryption AES-Twofish --filesystem=FAT --hash SHA512 --volume-type normal --size 1000000 --random-source /dev/urandom`, VeraCrypt container is created without a hitch. Error message, however, was not triggered this time upon mount. | :x: |

In summary, closest to the answer is using the default wizard (`veracrypt -t -c`) and selecting the "Linux EXT4" option to create EXT4-format VC container.

**Comment by davistan** [comment]
Thanks for the answer. Does this means that we are stuck with FAT if using automated command-line?

Just to share with you that I'm working on a docker setup where it will create a veracrypt container with  ext4. Ext4 is needed to ensure we don't hit the long file name (LFN) limitations.

**Comment by idrassi** [comment]
It turned out that this is a "bug" (or intentional limitation) inherited from TrueCrypt: the code that parses the command line ignored all filesystem values except FAT, although the filesystem formatting logic exists.

I have fixed this in this commit and I was able to check that it is working: https://github.com/veracrypt/VeraCrypt/commit/d53b48bdf02acca8972739f30dae547d0375c21b

One important note though: administrative right are needed to perform formatting other than FAT and so you will be prompted to enter the admin password (or your password for sudo). This is because VeraCrypt calls external tools (in your case mkfs.ext4) to perform the formatting and these tools can't work without admin rights. To avoid such prompt, the whole VeraCrypt command should be run with sudo.

Thank you for reporting this. I will publish binaries for the 1.17-BETA that include this. The release of the final 1.17 is delayed until we solve the new logo copyright issue (

**Comment by davistan** [comment]
Thanks again for the quick action. Looking forward for the beta release.

**Comment by idrassi** [comment]
I published VeraCrypt 1.17-BETA18 that contains this fix: https://veracrypt.codeplex.com/releases/view/619351
It also adds the possibility to create exFAT volumes which is an interesting alternative to FAT/NTFS/Ext4 since it enables large files (> 4GB), long file names and it is supported on all platforms (for Ubuntu, exfat-fuse is needed for read/write and exfat-utils is needed for formatting).

**Comment by davistan** [comment]
Thanks for the new release. Just confirm that the version still showing 1.16 for beta?
